edge/service/src/main/java/org/eclipse/ditto/edge/service/dispatching/EdgeCommandForwarderActor.java

```diff
@@ -151,25 +163,53 @@ public Receive createReceive() {
 
     private void forwardToThings(final Signal<?> thingSignal) {
         final ActorRef sender = getSender();
-        applySignalTransformation(thingSignal, sender)
-                .thenAccept(transformed -> {
-                    log.withCorrelationId(transformed)
-                            .info("Forwarding thing signal with ID <{}> and type <{}> to 'things' shard region",
-                                    transformed instanceof WithEntityId withEntityId ? withEntityId.getEntityId() :
-                                            null,
-                                    transformed.getType());
-
-                    if (!Signal.isChannelLive(transformed) &&
-                            !Signal.isChannelSmart(transformed) &&
-                            transformed instanceof Command<?> command &&
-                            isIdempotent(command)) {
-                        askWithRetryCommandForwarder.forwardCommand(command,
-                                shardRegions.things(),
-                                sender);
-                    } else {
-                        shardRegions.things().tell(transformed, sender);
-                    }
-                });
+        final CompletionStage<Signal<?>> signalTransformationCs = applySignalTransformation(thingSignal, sender);
+
+        scheduleTask(thingSignal, () -> signalTransformationCs.thenAccept(transformed -> {
+            log.withCorrelationId(transformed)
+                    .info("Forwarding thing signal with ID <{}> and type <{}> to 'things' shard region",
+                            transformed instanceof WithEntityId withEntityId ? withEntityId.getEntityId() :
+                                    null,
+                            transformed.getType());
+
+            if (!Signal.isChannelLive(transformed) &&
+                    !Signal.isChannelSmart(transformed) &&
+                    transformed instanceof Command<?> command &&
+                    isIdempotent(command)) {
+                askWithRetryCommandForwarder.forwardCommand(command,
+                        shardRegions.things(),
+                        sender);
+            } else {
+                shardRegions.things().tell(transformed, sender);
+            }
+        }));
+    }
+
+    private void scheduleTask(final Signal<?> signal,
```

Comment:
maybe rename signal var to thingSignal so readability is improved.

Reply:
Not only thing signals are passed into that method, but also policy signals, connectivity commands, etc.